### ISSUE TITLE
allow accepting a patch with local changes

### ISSUE THREAD

**Issue by gkatsev** [videojs/video.js#2263]
We should update contrib settings to allow us to accept a patch/pr with local changes without requiring us to make a new PR and accepting that.

**Comment by gkatsev** [comment]
Also, it should be able to clean up a previous branch and re-use it instead of forcing us to delete it manually and retry

**Comment by heff** [comment]
I'm not understanding what you mean. Can you give an example?

**Comment by gkatsev** [comment]
Yep, I just wanted to stub it out so we won't forget.
There are two things here. My original post and my second comment. Hopefully, the second one is more obvious.

For the first item, say you have a PR you want to accept but there's a typo in there. Currently, afaik, to accept the patch and also update the typo, you'd need to pull down the PR, create a new commit and then either accept the PR manually (perhaps by copying stuff from the `contrib.json`) or by submitting a new PR with the original changes and your changes and then accepting that new PR. It would be nice if there was a `contrib pr accept-edit` or something like that that staged the PR/patch locally, allowed you to submit more commits locally and then do a `contrib pr accept-local` which would then finish up the PR acceptance using the currently checked out local branch.

The second one has to do with `contrib` reusing the same branch always. For example, `contrib pr review` will try to checkout and use a branch called `review-gkatsev-branch-name`. If you then need to go do something else and want to go back to reviewing that original PR which since has changed, you won't be able to use `contrib` for it since the branch already exists. It probably should check for the existence of the branch and if it exists do a `reset --hard remote/branch-name` on it.

Hopefully, the above clears it up.

**Comment by heff** [comment]
Yeah, nice. Thanks.

For the first part, even for small changes I've used the `contrib feature modify` command, which takes changes to the contrib review branch and makes a new PR against the original submitter's PR. Once they pull in your PR their PR gets updated automatically. And then you can go from there. This helps people learn the standards and style you want to keep for the project, and keeps from giving the expectation that you'll just clean up after them. Have you tried that approach yet?

For the second part, that makes sense to me. Maybe a contrib review update command?

**Comment by gkatsev** [comment]
I guess as part of the [collaborator guide](https://github.com/videojs/video.js/issues/2250) we should make sure to outline what the various contrib commands are and what they do. Also, while in most cases `contrib feature modify` should be good, sometimes you just want to make the change and accept it, particularly, if it's a very small change like a typo where you don't want to wait on the back-and-forth with getting your changes merged into their PR.

Yeah, `contrib review update` could work. Though, I think it would just be nicer if the commands were either aware of the branch and auto-updated it, or it should just increment the branch name like if `foo` exists, then try `foo-1`. Though, this perhaps would require us to make more of a script that the `contrib.json` can use for checkout out branches instead (probably a pain to try to specificy this in that json format :)

**Comment by gkatsev** [comment]
Going to close this because I think we want to move away from contrib. Just adds a bit too much friction for other collaborators and I would like to grow the amount of collaborators that can and do merge things into master (and maybe release one day).
Plus, github has a new feature where you can push to a PR (unless the requester disabled it).
